wled00/FX.h

```diff
@@ -381,6 +381,26 @@ typedef struct Segment {
     byte     *data; // effect data pointer
     static uint16_t maxWidth, maxHeight;  // these define matrix width & height (max. segment dimensions)
 
+    typedef struct TemporarySegmentData {
```

Comment:
This is not bad at all, so this is just a suggestion, but wouldn't it be more maintainable to make an anonymous `struct CoreSegmentData` and use it both in `Segment` and `Transition`? All the members could still be accessed like usual (e.g. `seg.speed`) since the anonymous struct has no name. Copying could probably also be simplified from how it is implemented now in `saveSegenv()` as the whole struct could be copied in one line (not sure if it is possible to do that with anonymous structs though).